File: BIS.API/BIS.Web_App/projects/sharedlibrary/src/model/enum.ts
export class EnumBase {
    public get PermissionAction() {
        return PermissionAction;
      }
      public get PermissionItem() {
        return PermissionItem;
      }
      public get RoleType(){
        return RoleType;
      }
      public get NotificationType(){
        return NotificationType;
      }
}

export enum NotificationType {
  MasterData = 1,
  GenerateReport = 2
}

export enum PermissionItem {
    Dashboard = 'Dashboard',
    User = 'User',
    Sample = 'Sample',
    Supplier = 'Supplier',
    Buyer = "Buyer",
    Category = 'Category',
    Facility = 'Facility',
    Inventory = 'Inventory',
    UserRole = 'UserRole',
    Catalogue = 'Catalogue',
    BuyerPo = 'BuyerPo',
    SupplierOrder = 'SupplierOrder',
    InventoryInward='InventoryInward',
    ChannelSku='ChannelSku',
    CustomerOrder='CustomerOrder',
    Dispatch='Dispatch',
    UserMapping='UserMapping',
    Report='Report',
    InventoryQC = 'InventoryQC',
    InventoryPackaging = 'InventoryPackaging',
    InventoryProcess = 'InventoryProcess',
    InventoryTransfer = 'InventoryTransfer',
    Attribute = 'Attribute'
  }
  
  
  export enum PermissionAction {
    Read = 'Read',
    Create = 'Create',
    Update = 'Update',
    ReadAll = 'ReadAll',
    Delete = 'Delete',
  
  }

  export enum RoleType
  {
      Staff1 = "Staff1",
      Staff2 = "Staff2",
      G1Int = "G1Int",
      Colgs = "Colgs",
      ColInt = "ColInt",
      Bgs = "Bgs",
      Mggs = "Mggs",
      Goc = "Goc",
      Admin = "Admin",
      SuperAdmin = "SuperAdmin"
  }

File: BIS.Common/Enum/Enum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using Newtonsoft.Json.Converters;

namespace BIS.Common.Enum
{
    public class Enum
    {
        [JsonConverter(typeof(StringEnumConverter))]
        public enum PermissionItem
        {
            Dashboard = 1,
            MasterData,
            AddMasterData,
            SearchAndQuery,
            GenerateReport,
            SmartAnalysis,
            PredictiveModel,
            CdrDashboard,
            ImportExport,
            SavedNotes,
            NotesForMe,
            Facility,
            User,
            Role,
            PermissionVerification,
            Attribute
        }
        [JsonConverter(typeof(StringEnumConverter))]
        public enum PermissionAction
        {
            Read = 1,
            Create,
            Update,
            ReadAll,
            Delete
        }
        [JsonConverter(typeof(StringEnumConverter))]
        public enum RoleType
        {
            Staff1 = 1,
            G1Int,
            Colgs,
            ColInt,
            Bgs,
            Mggs,
            Goc,
            Admin,
            SuperAdmin
        }
        [JsonConverter(typeof(StringEnumConverter))]
        public enum NotificationType 
        { 
            MasterData = 1,
            GenerateReport
        }
        [JsonConverter(typeof(StringEnumConverter))]
        public enum Status
        {
            Created = 1,
            Progress,
            Approved,
            Rejected
        }
        [JsonConverter(typeof(StringEnumConverter))]
        public enum CategoryLoc : short
        {
            SourceLoc = 1,
            TypeOfLoc
        }
        public enum DaysMonthFilter
        {
            All = 1,
            Days30,
            Today,
            Months12
        }
    }
}
